XRTK-Core/Packages/com.xrtk.core/Providers/Controllers/Hands/HandRay.cs

```diff
@@ -0,0 +1,106 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+using XRTK.Definitions.Utilities;
+using XRTK.Utilities.Physics;
+
+namespace XRTK.Providers.Controllers.Hands
+{
+    public class HandRay
```

Comment:
Might be best to introduce an IHandRay interface right away

As I'm doing for MRTK.
https://github.com/microsoft/MixedRealityToolkit-Unity/pull/6973/files#diff-996cc5a57a714af0f8c171ab2f067e54

Reply:
I honestly think we need to just provide a better way to specify custom Ray implementations and not be specific to hands.

Reply:
Well, that's outside the scope for this PR, But I agree, until it's fully implemented, this should just be an interface and not a class.

Reply:
This PR is for the final feature. If anything I don't want a feature to go in until it's ready.